lib/matplotlib/ticker.py

```diff
@@ -2698,6 +2698,10 @@ def get_log_range(lo, hi):
                     ticklocs.extend(subs * decade)
         else:
             ticklocs = decades
+            # if there is not enough ticks to show on the graph
+            if len(ticklocs) <= 2:
+                subs = np.linspace(vmin, vmax, 5)
```

Comment:
Two questions:

1. Why is this 5? Shouldn't 3 be sufficient (or even 3 - ticklocs)?
2. Can you convince me that this is the right fix? I'm not quite into the locator code, but from a quick look I have the impression that if there are less than two decades, there should be something similar to the above `subs = np.arange(2.0, base)` handling, i.e. place labels on all the minor ticks (or maybe even on a subset of them) - but not on "arbitrary" locations.